hvac/api/secrets_engines/transit.py

```diff
@@ -861,6 +868,9 @@ def sign_data(
                     allowed_types=transit_constants.ALLOWED_SALT_LENGTHS.pattern,
                 )
             )
+        
+        hash_input = None if batch_input is not None else hash_input
```

Comment:
To ensure compatibility, I included the following line to set the value of `hash_input` to `None` when `batch_input` is set, as both values cannot be simultaneously used.

Reply:
Thanks for pointing this out. I am thinking about whether we should continue to have `hash_input` be a required parameter (argument) in the function, or we should default it to `None`, and then have code to raise an exception if `hash_input is None and batch_input is None`.

Basically, I am thinking it might be an awkward user experience when using `batch_input` if you are forced to explicitly set `hash_input=None`.

Since you would presumably be using this exact scenario, what do you think about that?

Reply:
I think it would be better set both parameters to None by default. In case the user fails to provide a value for any of the parameters, an exception can be raised with message guiding the user to set the necessary values.